packages/util-crypto/src/mnemonic/toLegacySeed.ts

```diff
@@ -24,10 +24,16 @@ import { mnemonicValidate } from './validate';
  * }
  * ```
  */
-export function mnemonicToLegacySeed (mnemonic: string, password = '', onlyJs = false): Uint8Array {
+export function mnemonicToLegacySeed (mnemonic: string, password = '', onlyJs = false, byteLength?: number): Uint8Array {
   assert(mnemonicValidate(mnemonic), 'Invalid bip39 mnemonic specified');
 
-  return isReady() && !onlyJs
-    ? bip39ToSeed(mnemonic, password)
-    : mnemonicToSeedSync(mnemonic, password).subarray(0, 32);
+  if (byteLength && byteLength === 32) {
```

Comment:
Would suggest an assert on the 32 | 64, e.g. `assert([32, 64].includes)byteLength), `Invalid seed length ${byteLength}, expected 32 or 64`);